src/client/common/experiments/service.ts

```diff
@@ -120,6 +125,7 @@ export class ExperimentService implements IExperimentService {
                 return false;
             }
             case 'optIn': {
+                await this.experimentationService.isCachedFlightEnabled(experiment);
```

Comment:
Regardless of whether user has opted in manually, we need to check with the experiment service.
This way if a user falls into an experiment, the telemetry will line up with experiments on server.